xmodule/modulestore/__init__.py

```diff
@@ -313,13 +324,54 @@ def send_pre_publish_signal(self, bulk_ops_record, course_id):
         if signal_handler and bulk_ops_record.has_publish_item:
             signal_handler.send("pre_publish", course_key=course_id)
 
+    def create_catalog_data_for_signal(self, course_id):
+        """
+        Creates a CourseCatalogData object from a course id by fetching from the module store
+
+        Arguments:
+            course_id (CourseKey): id of the course to process
+
+        Returns:
+            CourseCatalogData
+        """
+        # Most of this is copied from CourseOverview. We cannot import it
+        # because it would lead to circular imports
+        course_from_store = self.get_course(course_id)
+        number = number_for_course_location(course_from_store.location)
+        catalog_visibility = course_from_store.catalog_visibility
+
+        # see CourseDetail.fetch_about_attribute
+        effort_usage_key = course_id.make_usage_key('about', 'effort')
+        try:
+            effort = self.get_item(effort_usage_key).data
+        except ItemNotFoundError:
+            effort = None
+        return CourseCatalogData(
+            course_key=course_id.for_branch(None),
+            name=course_from_store.display_name,
+            org=course_from_store.location.org,
+            number=number,
+            schedule_data=CourseScheduleData(
+                start=course_from_store.start,
+                pacing='self' if course_from_store.self_paced else 'instructor',
+                end=course_from_store.end,
+                enrollment_start=course_from_store.enrollment_start,
+                enrollment_end=course_from_store.enrollment_start,
+            ),
+            effort=effort,
+            hidden=catalog_visibility in ['about', 'none'] or course_from_store.id.deprecated,
+            invitation_only=course_from_store.invitation_only,
+        )
+
     def send_bulk_published_signal(self, bulk_ops_record, course_id):
         """
         Sends out the signal that items have been published from within this course.
         """
         if self.signal_handler and bulk_ops_record.has_publish_item:
             # We remove the branch, because publishing always means copying from draft to published
             self.signal_handler.send("course_published", course_key=course_id.for_branch(None))
+            catalog_data = self.create_catalog_data_for_signal(course_id)
```

Comment:
This event is only for one course but as part of bulk operations, which seems to be what happens when running basic Studio locally.